src/theory/strings/skolem_cache.cpp

```diff
@@ -312,7 +317,15 @@ Node SkolemCache::mkSkolemFun(SkolemFunId id, TypeNode tn, Node a, Node b)
   }
   NodeManager* nm = NodeManager::currentNM();
   SkolemManager* sm = nm->getSkolemManager();
-  Node k = sm->mkSkolemFunction(id, tn, cacheVals);
+  Node k;
+  if (tn.isNull())
+  {
+    k = sm->mkSkolemFunction(id, cacheVals);
+  }
+  else
+  {
+    k = sm->mkSkolemFunctionTyped(id, tn, cacheVals);
+  }
```

Comment:
The interface for `mkSkolemFunTyped` here is different from `mkSkolemFunctionTyped`, which does not handle a `null` type. I prefer calling `sm->mkSkolemFunction` directly from `mkSkolemFun`.

Reply:
Are you suggesting to allow null type in `SkolemManager::mkSkolemFunction`? Or what is the suggestion?